course/include/Matrix3.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <iostream>
+#include <cmath>
+#include "Vector3.h"
+
+namespace ekumen
+{
+
+    class Matrix3
+    {
+    public:
+        Matrix3();
+        explicit Matrix3(const Vector3 &f1, const Vector3 &f2, const Vector3 &f3);
+        Matrix3(const std::initializer_list<double> &list);
+        Matrix3(const std::initializer_list<double> &r1, const std::initializer_list<double> &r2, const std::initializer_list<double> &r3);
+        Matrix3(const Matrix3 &m);
+
+        double det() const;
+        const Vector3 row(const int index) const;
+        Vector3 &row(const int index);
+        const Vector3 col(const int index) const;
+
+        Matrix3 mul(const Matrix3 &m2) const;
+        Matrix3 inverse() const;
+
+        Matrix3 &operator+=(const Matrix3 &q);
+        Matrix3 &operator-=(const Matrix3 &q);
+
+        //operator overload
+        const Vector3 &operator[](const int &index) const;
```

Comment:
Please also overload `operator=()`, both for copy and move.

Reply:
Done